### ISSUE TITLE
Link Control select and edit link suggestion without auto-apply

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#47543]
Currently selecting a search suggestion within the Link Control component will automatically apply that link and close the UI.

@joedolson is suggesting that instead the suggestion should be selected but the link not applied until you have "committed" your changes. Presumanly by hitting "Apply".

I'm going to ask for input from @WordPress/gutenberg-design as this will be a significant change to the UX which may require additional input.

-----

> At the moment, there's no way to select a link from a search & adjust its settings in a single step. Since selecting the link creates the link and closes the panel, you have to re-select the link and edit to change its settings. This is a minor annoyance for mouse users, but fairly significant if you're navigating by keyboard. 

_Originally posted by @joedolson in https://github.com/WordPress/gutenberg/issues/47310#issuecomment-1399557772_

**Comment by jameskoster** [comment]
> Since selecting the link creates the link and closes the panel, you have to re-select the link and edit to change its settings. 

I do see how this would be frustrating, but it's only applicable to internal links, for which it seems less likely you'd need to adjust any of the settings?

It seems a bit of a shame to introduce the extra click. But in practice perhaps it's not so bad? I don't have a strong feeling and wouldn't object to trying this. Keen to hear thoughts from others.

**Comment by getdave** [comment]
I think maybe we could make a small change here as a middle ground to help. When a user selects a suggestion we allow it to auto-apply _but_ we wouldn't close the Link UI. That means if a user has additional edits they can tab 1 stop to "edit" and then make the changes.

Currently we close the Link UI entirely.

**Comment by jameskoster** [comment]
Hmm I think that might undermine the apply button a bit. Closing the panel works because it's more obvious. Imo it wouldn't hurt to try the original suggestion (requiring a click of the Apply button) to see how it feels, then gather more feedback.

**Comment by getdave** [comment]
I'm hopeful that https://github.com/WordPress/gutenberg/issues/50891 will address this in time.
